### PR TITLE
Python package set: major updates

### PR DESCRIPTION
**Pull request by FRidh**
###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/development/python-modules/ftfy/default.nix [hunk 1]
```diff
@@ -11,7 +11,7 @@
 buildPythonPackage rec {
   pname = "ftfy";
 
-  version = "4.4.3";
+  version = "5.4.0";
   # ftfy v5 only supports python3. Since at the moment the only
```
Comment (Mic92) [inline pkgs/development/python-modules/ftfy/default.nix]: Does the comment still apply?

File: pkgs/development/python-modules/netcdf4/default.nix [hunk 2]
```diff
@@ -4,13 +4,13 @@
 buildPythonPackage rec {
   pname = "netCDF4";
   name = "${pname}-${version}";
-  version = "1.3.1";
+  version = "1.4.0";
 
   disabled = isPyPy;
 
   src = fetchurl {
```
Comment (Mic92) [inline pkgs/development/python-modules/netcdf4/default.nix]: fetchPypi

**Review by timokau — COMMENTED** [review]
Apparently rpy2 2.9.x doesn't support python2 anymore:

```
rpy2 is no longer supporting Python < 3. Consider using an older rpy2 release when using an older Python release.
```

So we should probably keep a 2.8.x version around. And mark 2.9.x as python3 only.

File: pkgs/development/python-modules/notebook/default.nix [hunk 3]
```diff
@@ -24,11 +24,11 @@
 
 buildPythonPackage rec {
   pname = "notebook";
-  version = "5.4.1";
+  version = "5.5.0";
```
Comment (timokau) [inline pkgs/development/python-modules/notebook/default.nix]: This breaks the build:

```
ERROR: Failure: ImportError (No module named selenium.webdriver.common.by)
```

Comment (timokau) [reply pkgs/development/python-modules/notebook/default.nix]: Adding `selenium` to `checkInputs` fixes it (selenium changelog: https://github.com/jupyter/notebook/blob/master/docs/source/changelog.rst).

Comment (Mic92) [reply pkgs/development/python-modules/notebook/default.nix]: I don't think we should test selenium at all - those tests are too brittle in my experience, the hole test suite could be deleted before building: https://github.com/jupyter/notebook/tree/ebe01762eb3e93b2c3f525eceb27ca6dd9611000/notebook/tests/selenium

**Comment by flokli** [comment]
The "python-dateutil: 1.5 -> 2.7.3" commit 07679a9 broke `khal` as well:

```
Processing ./khal-0.9.9-py3-none-any.whl
Collecting python-dateutil<=2.6.1 (from khal==0.9.9)
  Could not find a version that satisfies the requirement python-dateutil<=2.6.1 (from khal==0.9.9) (from versions: )
No matching distribution found for python-dateutil<=2.6.1 (from khal==0.9.9)
builder for '/nix/store/ln6clsi7qpfa90kr054lrlf9lm33qlkx-khal-0.9.9.drv' failed with exit code 1
```

**Comment by dotlambda** [comment]
@flokli khal should be fixed by https://github.com/NixOS/nixpkgs/pull/42782

**Comment by lopsided98** [comment]
`buildbot-pkg` needs to be excluded from these bulk updates unless you are going to update the rest of buildbot. Updating `buildbot-pkg` implicitly updates all the buildbot plugins, causing them to have hash mismatches. #38483 caused this same breakage before.

**Comment by mbode** [comment]
@FRidh You probably did not mean to bump the version in _pkgs/development/python-modules/dateutil/1_5.nix_ either, right?

**Comment by mbode** [comment]
@dotlambda I don't know, just randomly stumbled over it. I guess if it isn't, it should be deleted.

**Comment by dotlambda** [comment]
> I guess if it isn't, it should be deleted.

Done in d7cce082b09.

**Pull request merged**